src/include/setupUI.js

```diff
@@ -536,13 +537,16 @@ WMEAC.connectAdvancedClosureDialogHandlers = function ()
                 alert("Can't apply closures.\nPlease, check all parameters.");
                 return;
             }
-            if (W.selectionManager.getSelectedFeatures().length==0 || W.selectionManager.getSelectedFeatures()[0].model.type!="segment")
+            const m = W.selectionManager.getSelectedDataModelObjects();
+            if (m.length==0 || m[0].type != "segment")
             {
                 alert("Please, select segment(s) before.");
                 return;
             }
             if (W.selectionManager.getSelectedFeatures().every(function (e) {
```

Comment:
I would modify this to W.selectionManager.getSelectedDataModelObjects().every...
and
const segid = e.id;

Reply:
Looks good. I will make those changes.

Reply:
For some reason, e.id is null here. e.attributes.id would work.

Reply:
changed to use getSelectedDataModelObjects()[]  .atttributes.id here